lib/puppet/functions/vault_lookup/lookup.rb

```diff
@@ -1,10 +1,15 @@
 Puppet::Functions.create_function(:'vault_lookup::lookup') do
   dispatch :lookup do
     param 'String', :path
-    param 'String', :vault_url
+    optional_param 'String', :vault_url
```

Comment:
Maybe this is s good opportunity to change the type to `Stdlib::HTTPUrl`?

Reply:
I'd rather move that to a seperate PR, since it would be a breaking change as we'd have to add a new requirement on the stdlib module